### PR TITLE
Daniandrew/appeals 62757

### PR DESCRIPTION
**Pull request by dcoleman21**
Resolves [Implement Caseflow Remediation Audit History](https://jira.devops.va.gov/browse/APPEALS-62757)

# Description

- migration for CreateEventRemediationAudit
- EventRemediationAudit table created
- EventRemediationAudit model added (may need to check the associations)
- Added logic for creating a EventRemediationAudit in veteran_record_remediation_service class 
- Added logic for creating a EventRemediationAudit in duplicate_person_remediation_service class
- Added tests for all 

## Acceptance Criteria
- [x] Code compiles correctly

# Backend
## Database Changes
*Only for Schema Changes*

* [x] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [x] Update column comments; include a "PII" prefix to indicate definite or potential [PII data content](https://github.com/department-of-veterans-affairs/appeals-team/blob/main/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii)
* [x] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [ ] For queries using raw sql was an explain plan run by System Team
* [x] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [x] Add `belongs_to` for associations to enable the [schema diagrams](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/schema_diagrams) to be automatically updated
* [ ] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

# Best practices
## Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [x] No new code climate issues added

### Error Handling
- [x] Are errors being caught and handled gracefully?
- [x] Are appropriate error messages being displayed to users?
- [ ] Are critical errors being reported to an error tracking system (e.g., Sentry, ELK)?
- [ ] Are unhandled exceptions being caught at the application level ?

### CODE REVIEW

**Review by raymond-hughes — DISMISSED** [review]

**Review by raymond-hughes — DISMISSED** [review]

File: app/services/remediations/duplicate_person_remediation_service.rb [hunk 1]
```diff
@@ -40,4 +46,18 @@ def find_and_update_records
       false
     end
   end
+
+  def add_remediation_audit(remediated_record:, before_data:, after_data:)
+    EventRemediationAudit.create!(
+      event_record: @event_record,
+      remediated_record_type: remediated_record.class.name,
+      remediated_record_id: remediated_record.id,
+      # remediated_record: remediated_record,
```
Comment (NoelleAd) [inline app/services/remediations/duplicate_person_remediation_service.rb:55]: Can we remove commented out code or contextualize why it's left in?

File: app/services/remediations/veteran_record_remediation_service.rb [hunk 2]
```diff
@@ -51,11 +52,29 @@ def update_records!(collections, file_number)
...
       end
     end
   end
 
+  def add_remediation_audit(remediated_record:, before_data:, after_data:)
+    EventRemediationAudit.create!(
+      event_record: @event_record,
+      remediated_record_type: remediated_record.class.name,
+      remediated_record_id: remediated_record.id,
+      # remediated_record: remediated_record,
```
Comment (NoelleAd) [inline app/services/remediations/veteran_record_remediation_service.rb:69]: Consider removing this line.

**Review by ryanpmessner — DISMISSED** [review]
The optimization could wait till a later PR if you wanna get this in, but it should probably be done if this is going to happen with any frequency -- the other is just a question about the intentionality of a factory change

File: spec/factories/sequences.rb [hunk 3]
```diff
@@ -42,10 +42,11 @@ def shortened_time
   end
 
   # User factory
-  sequence :css_id do |n|
-    FactoryBot.sequence_by_name(:css_id).rewind if n == 999
-    shortened_time.concat(format("%<n>03d", n: n))
-  end
+  # sequence :css_id do |n|
```
[the remark this answers is not in the document]
Comment (dcoleman21) [reply spec/factories/sequences.rb:45]: Yeah I am not sure how that got into this PR but essentially had to comment out the code and add the lines below it.

[the remark this answers is not in the document]
Comment (dcoleman21) [reply spec/factories/sequences.rb]: We reverted to match main

File: app/services/remediations/duplicate_person_remediation_service.rb (hunk 1 above)
Comment (dcoleman21) [reply app/services/remediations/duplicate_person_remediation_service.rb]: Removed it.

**Review by raymond-hughes — APPROVED** [review]

**Pull request merged**